### DISCUSSION TITLE
Self hosting Supabase + Chatbot-UI Documentation

### DISCUSSION THREAD

**Discussion by im-calvin** [mckaywrigley/chatbot-ui#1289] [Help]
Hi all, I've been working on self hosting this entire project for a little while and I just thought to document the things I've had to do in case any of you guys find it helpful.

I followed the instructions [here](https://supabase.com/docs/guides/self-hosting) to self host Supabase. Although whenever it says to restart the database, I just ran the uninstall and reinstall scripts. 

For creating JWT_SECRETS, use Bitwarden or, what I did, was create a random Supabase project and use those keys and then delete the project. If u run into `JWSError JWS Signature` problems, your JWT_SECRET is probably not a valid JWT. 

For `db-url`, it is in this format: `postgresql://postgres:PASSWORD@IP_ADDR:5432/postgres` if you follow the default configs. Use this with `supabase db push --db-url` in the client (this) codebase.

For custom SMTP, you have to modify the values in the `.env` file in the database, note that `SMTP_SENDER_NAME, SMTP_ADMIN_EMAIL` are whatever you want it to be. For the signup email, I couldn't figure out how to specify the address (and more importantly, the port) for wherever it redirects to you when you click signup on Chatbot-UI, so I just reverse-proxied with nginx the database to go from port 8000 to default.

To edit the default email template (which I didn't do), you seem to have to put your own HTML file from the database side. 

Azure has a bunch of problems in Chatbot-UI, it seems like more work needs to be done on this end, also with `.env` API variables. If they are set, during onboarding those field should be closed and there's a problem with selected models / talking to Azure models with environment API keys.

To allow file uploading and not get a weird `413 file size` error, in the database nginx config, I had to enable `server_names_hash_bucket_size 64` and it worked. Not sure why but there's a possible solution for you.

There's some problems with infinite load spinners, solution being refreshing the page. Not really a big problem, but would be nice fix for UX

I've currently been running into an issue where `operator does not exist: extensions.vector <=> extensions.vector` in the database when I try and talk to a document that I uploaded, and I think that the database doesn't have `pgvector` installed properly: on the dashboard when I toggle the vector extension, it gives me a weird undefined error. I've tracked it down to the `match_file_items_local` function in Postgres.

**Comment by im-calvin** [comment]
Solved the extensions.vector thing by following this thread: https://github.com/supabase/supabase/issues/13337

**Comment by alexnechifor** [comment]
Hello! I also have a problem with self deployed supabase and FE on azure. I'm getting CORS error while trying to upload the file. I fixed the problem with PG Vector. Any ideas? I'm also running a nginx proxy.

```
Access to fetch at 'https://example/storage/v1/object/files' from origin 'https://example' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

**Reply by im-calvin** [reply]
My best guess is that you need to add your origin to `next.config.js`

Example:

```
  experimental: {
    serverActions: {
      allowedOrigins: [
        "localhost:3000",
        "https://example"
      ]
    },
    serverComponentsExternalPackages: ["sharp", "onnxruntime-node"]
  }
```
